mining.go

```diff
@@ -216,6 +216,38 @@ func txPQByStakeAndFee(pq *txPriorityQueue, i, j int) bool {
 	return pq.items[i].feePerKB > pq.items[j].feePerKB
 }
 
+// txPQByStakeAndFeeAndThenPriority sorts a txPriorityQueue by stake priority,
+// followed by fees per kilobyte, and then if the transaction type is regular
+// or a revocation it sorts it by priority.
+func txPQByStakeAndFeeAndThenPriority(pq *txPriorityQueue, i, j int) bool {
+	// Sort by stake priority, continue if they're the same stake priority.
+	cmp := compareStakePriority(pq.items[i], pq.items[j])
+	if cmp == 1 {
+		return true
+	}
+	if cmp == -1 {
+		return false
+	}
+
+	bothAreLowStakePriority :=
+		txStakePriority(pq.items[i].txType) == regOrRevocPriority &&
```

Comment:
This is another reason why doing the check directly as I had suggested would be better since it avoid the need to figure out the priority twice.

```Go
	iStakePriority := txStakePriority(pq.items[i].txType)
	jStakePriority := txStakePriority(pq.items[j].txType)
	if iStakePriority > jStakePriority {
		return true
	}
	if iStakePriority < jStakePriority {
		return false
	}

	bothAreLowStakePriority := iStakePriority == regOrRevocPriority && jStakePriority == regOrRevocPriority
	...
```